Filter/AbstractDateFilter.php

```diff
@@ -165,6 +165,24 @@ public function getRenderSettings()
             $name .= '_range';
         }
 
+        // NEXT_MAJOR: Remove this line when drop Symfony <2.8 support
+        if (method_exists('Symfony\Component\Form\AbstractType', 'getBlockPrefix')) {
+            switch ($name) {
+                case 'sonata_type_filter_date':
+                    $name = 'Sonata\AdminBundle\Form\Type\Filter\DateType';
+                    break;
+                case 'sonata_type_filter_date_range':
+                    $name = 'Sonata\AdminBundle\Form\Type\Filter\DateRangeType';
+                    break;
+                case 'sonata_type_filter_datetime':
+                    $name = 'Sonata\AdminBundle\Form\Type\Filter\DateTimeType';
+                    break;
+                case 'sonata_type_filter_datetime_range':
+                    $name = 'Sonata\AdminBundle\Form\Type\Filter\DateTimeRangeType';
+                    break;
+            }
+        }
+
```

Comment:
Not sure if we can improve this with an inflector class. We can't convert everything from snake to camel case.